src/pharmpy/tools/structsearch/drugmetabolite.py

```diff
@@ -1,69 +1,84 @@
+        and len(metabolite_functions) != 0
+    ):
+        raise NotImplementedError(
+            'Metabolite structure transformations not yet supported when already present in the mode. '
+            'Try removing and only applying peripheral transformations'
+        )
 
-    Parameters
-    ----------
-    model : Model
-        Base model for search.
-    route : str
-        Type of administration. Currently 'oral', 'iv' and 'ivoral'
+    met_type_models = []
+    for eff, func in metabolite_functions.items():
+        candidate_model = func(model)
+        candidate_model = candidate_model.replace(description='_'.join(eff))
+        met_type_models.append(candidate_model)
 
-    Returns
-    -------
-    List[Model]
-        A list of candidate models.
-    """
-    models = []
-    if route not in ('oral', 'ivoral'):
-        presystemic_option = [False]
-    else:
-        presystemic_option = [False, True]
-    for presystemic in presystemic_option:
-        candidate_model = add_metabolite(model, presystemic=presystemic)
-        candidate_model = set_name(candidate_model, 'presystemic')
+    if len(met_type_models) == 0:
+        met_type_models = [model]
 
-        candidate_model = candidate_model.replace(parent_model='base_metabolite')
-        if presystemic:
+    base_model = None
+    # Model require metabolite compartment to extract peripherals functions for metabolite comp
+    if len(peripheral_mfl_statements) != 0:
+        peripherals_functions = funcs(
+            met_type_models[0], peripheral_mfl_statements, structsearch_metabolite_features
+        )
+        for metabolite_model in met_type_models:
+            previous_model = metabolite_model
+            for eff, func in peripherals_functions.items():
+                candidate_model = func(previous_model)
+                candidate_model = candidate_model.replace(
+                    description=candidate_model.description + f';MET_{eff[0]}({str(eff[1])})'
+                )
+                candidate_model, model_index = set_index(candidate_model, model_index)
+                if previous_model.name == metabolite_model.name:
+                    candidate_model = candidate_model.replace(parent_model="_TEMP")
+                    if base_model is None and metabolite_model.description.endswith("BASIC"):
+                        base_model = candidate_model.replace(parent_model=candidate_model.name)
+                else:
+                    candidate_model = candidate_model.replace(parent_model=previous_model.name)
+                models.append(candidate_model)
+                previous_model = candidate_model
+    else:
+        for m in met_type_models:
+            candidate_model, model_index = set_index(m, model_index)
+            if base_model is None and candidate_model.description.endswith("BASIC"):
+                base_model = candidate_model.replace(parent_model=candidate_model.name)
```

Comment:
I find this part a little confusing, is it not possible to go through the functions in one for-loop? Otherwise, perhaps it would be a bit clearer if you write a comment on why this complexity is needed?